modules/programs/neomutt.nix

```diff
@@ -0,0 +1,327 @@
+        description = "Specify the action to take.";
+      };
+    };
+  };
+
+  yesno = x: if x then "yes" else "no";
+  setOption = n: v: if v == null then "unset ${n}" else "set ${n}=${v}";
+  escape = replaceStrings ["%"] ["%25"];
+
+  accountFilename = account:
+    config.xdg.configHome + "/neomutt/" + account.name;
+
+  genCommonFolderHooks = account: with account;
+    let
+      passCmd = concatStringsSep " " passwordCommand;
+    in
+      {
+        from = "'${address}'";
+        realname = "'${realName}'";
+        spoolfile = "'+${folders.inbox}'";
+        record = if folders.sent == null then null else "'+${folders.sent}'";
+        postponed = "'+${folders.drafts}'";
+        trash = "'+${folders.trash}'";
+      };
+
+  mtaSection = account: with account;
+    if neomutt.sendMailCommand != null then {
+      sendmail = "'${neomutt.sendMailCommand}'";
+    } else let
+      smtpProto = if smtp.tls.enable then "smtps" else "smtp";
+      smtpBaseUrl = "${smtpProto}://${escape userName}@${smtp.host}";
+    in {
+        smtp_url = "'${smtpBaseUrl}'";
+        smtp_pass = "'`${passCmd}`'";
+    };
+
+  genMaildirAccountConfig = account: with account;
+    let
+      folderHook =
+        mapAttrsToList setOption (
+          genCommonFolderHooks account
+          // { folder = "'${account.maildir.absPath}'"; }
+        )
+        ++ optional (neomutt.extraConfig != "") neomutt.extraConfig;
+    in
+      ''
+        ${concatStringsSep "\n" folderHook}
+      ''
+      ;
+
+  registerAccount = account: with account;
+    ''
+    # register account ${name}
+    mailboxes "${account.maildir.absPath}/${folders.inbox}"
+    folder-hook ${account.maildir.absPath}/ " \
+        source ${accountFilename account} "
+    '';
+
+    mraSection = account: with account;
+     if (account.maildir != null) then
+       genMaildirAccountConfig account
+      else
+        throw "Only maildir is supported at the moment"
+     ;
+
+      optionsStr = attrs:
+        concatStringsSep "\n" (mapAttrsToList setOption attrs);
+
+      # TODO remove
+      boolStr = yesno;
+
+        sidebarSection = ''
+          # Sidebar
+          set sidebar_visible = yes
+          set sidebar_short_path = ${yesno cfg.sidebar.shortPath}
+          set sidebar_width = ${toString cfg.sidebar.width}
+          set sidebar_format = '${cfg.sidebar.format}'
+        '';
+
+        bindSection =
+          concatMapStringsSep
+            "\n"
+            (bind: "bind ${bind.map} ${bind.key} \"${bind.action}\"")
+            cfg.binds;
+
+        macroSection =
+          concatMapStringsSep
+            "\n"
+            (bind: "macro ${bind.map} ${bind.key} \"${bind.action}\"")
+            cfg.macros;
+
+        mailCheckSection = ''
+          set mail_check_stats
+          set mail_check_stats_interval = ${toString cfg.checkStatsInterval}
+        '';
+
+
+        notmuchSection = account: with account;
+        ''
```

Comment:
Either move the `''` up to the previous line or indent the whole string two more spaces.